lualibs/go/x509/api.go

```diff
@@ -0,0 +1,63 @@
+package x509
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	lua "github.com/yuin/gopher-lua"
+)
+
+func pushErr(l *lua.LState, msg string) int {
+	base := l.GetTop()
+	l.Push(lua.LNil)
+	l.Push(lua.LString(msg))
+	return l.GetTop() - base
+}
+
+func validateCertificate(l *lua.LState) int {
+	certPEM := l.CheckString(1)
+	block, _ := pem.Decode([]byte(certPEM))
+	if block == nil {
+		return pushErr(l, "failed to parse certificate")
+	}
+
+	_, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return pushErr(l, err.Error())
+	}
+
+	l.Push(lua.LTrue)
+	return 1
+}
+
+func validateKey(l *lua.LState) int {
+	keyPEM := l.CheckString(1)
```

Comment:
Do we need to test for `pem.Decode([]byte("")` in the event that no parameter is passed from the luaState? Granted this should never happen based on the Lua code you added from the custom validator so this might be overkill in my thinking.

Note: This is the same in `validateCertificate()`